scico/numpy/util.py

```diff
@@ -222,6 +221,36 @@ def is_nested(x: Any) -> bool:
     return isinstance(x, (list, tuple)) and any([isinstance(_, (list, tuple)) for _ in x])
 
 
+def broadcast_nested_shapes(
+    shape_a: Union[Shape, BlockShape], shape_b: Union[Shape, BlockShape]
+) -> Union[Shape, BlockShape]:
+    r"""Compute the result of applying a broadcasting binary operator to
+    (block) arrays with (possibly nested) shapes `shape_a` and `shape_b`.
+
+    Args:
+        shape_a: First array shape.
+        shape_b: Second array shape.
+
+    Returns:
+        A (possibly nested) shape tuple.
+
+    Example:
+        >>> broadcast_nested_shapes(((1, 1, 3), (2, 3, 4)), ((3,), (2, 1, 4)))
+        ((1, 1, 3), (2, 3, 4))
```

Comment:
Perhaps choose an example that doesn't just give a result of the same shape as the first argument?